voluptuous/tests/tests.py

```diff
@@ -557,3 +557,61 @@ def fn(arg):
         return "hello"
 
     assert_raises(Invalid, fn, 1)
+
+def test_number_validation_with_string():
+    """ test with Number with string"""
+    schema = Schema({"number" : Number(precision=6, scale=2)})
+    try:
+        schema({"number": 'teststr'})
+    except MultipleInvalid as e:
+        assert_equal(str(e),
+                     "Value must be a number for dictionary value @ data['number']")
+    else:
+        assert False, "Did not raise Invalid for String"
+
+def test_number_validation_with_invalid_precision_scale():
+    """ test with Number with invalid precision and scale"""
+    schema = Schema({"number" : Number(precision=6, scale=2)})
+    try:
+        schema({"number": 123456.712})
+    except MultipleInvalid as e:
+        assert_equal(str(e),
+                     "Precision must be smaller than 6, and Scale must be smaller than 2 for dictionary value @ data['number']")
+    else:
+        assert False, "Did not raise Invalid for String"
+
+def test_number_validation_with_valid_number():
+    """ test with Number with valid precision and scale"""
+    schema = Schema({"number" : Number(precision=6, scale=2)})
```

Comment:
But the point of having a separate `Number` validator is to tackle these kind of issues only. If we want 1234, directly to be accepted then instead of using `Number` we can use `int` which is already supported by voluptuous.